src/utils/mixins.js

```diff
@@ -69,6 +73,34 @@ const usernameFormatter = {
   },
 };
 
+/**
+ * A mixin to generically fetch display properties
+ * from a girder_jobs.Job schema'd json object.
+ */
+const jobFormatter = {
+  methods: {
+    progressAsNumber(progress) {
+      if (!progress) {
+        return 100;
+      }
+      return 100 * (progress.current / progress.total);
+    },
+    formatJob(job) {
+      const statusDef = Object.assign({ text: 'Unknown' }, jobstatus.getByValue(job.status));
+      return Object.assign({
+        statusText: statusDef.text,
+        statusColor: statusDef.color,
+        statusTextColor: statusDef.textColor || 'white',
+        statusIcon: statusDef.icon,
+        updateString: moment(job.updated).format('dddd, MMMM D, YYYY @ h:mm a'),
+        progressNumber: this.progressAsNumber(job.progress),
+        indeterminate: statusDef.indeterminate,
+        spin: statusDef.spin,
+      }, job);
```

Comment:
I generally prefer `...` for multi-value assignment, but I don't feel strongly about it.

Reply:
:+1: